torch/csrc/cuda/memory_snapshot.cpp

```diff
@@ -117,6 +147,8 @@ void _record_memory_history(
     when = c10::cuda::CUDACachingAllocator::RecordContext::STATE;
   }
   at::globalContext().lazyInitCUDA();
+  c10::call_once(
```

Comment:
I think this can all go in _initRecordAnnotations() by just putting the body inside a lambda?

Reply:
Sounds good, updated.